universal-application-tool-0.0.1/app/services/question/types/QuestionDefinition.java

```diff
@@ -244,4 +249,13 @@ public boolean equalsIgnoreId(Object other) {
     }
     return false;
   }
+
+  private boolean questionTextAndHelpTextContainsRepeatedEntityNameFormatString() {
+    boolean textMissingFormatString =
+        questionText.translations().values().stream().anyMatch(text -> !text.contains("$this"));
+    boolean helpTextMissingFormatString =
+        questionHelpText.translations().values().stream()
+            .anyMatch(helpText -> !helpText.contains("$this"));
+    return !textMissingFormatString && !helpTextMissingFormatString;
```

Comment:
I think I agree with you, it would be more flexible for this to be "or", but we may need to change `anyMatch` to `allMatch`. I'm worried about a case where a translation is missing a reference to $this and we don't catch it

Reply:
The text is missing the format string if any translations do not contain "$this". 

I think that's what is implemented. I'm adding a test to make sure though.